zeebe/zb-db/src/main/java/io/camunda/zeebe/db/impl/rocksdb/transaction/TransactionalColumnFamily.java

```diff
@@ -208,43 +221,49 @@ public void whileEqualPrefix(
 
   @Override
   public void deleteExisting(final KeyType key) {
-    ensureInOpenTransaction(
-        transaction -> {
-          columnFamilyContext.writeKey(key);
-          assertKeyExists(transaction);
-          transaction.delete(
-              transactionDb.getDefaultNativeHandle(),
-              columnFamilyContext.getKeyBufferArray(),
-              columnFamilyContext.getKeyLength());
-        });
+    try (final var timer = metrics.measureDeleteLatency()) {
+      ensureInOpenTransaction(
+          transaction -> {
+            columnFamilyContext.writeKey(key);
+            assertKeyExists(transaction);
+            transaction.delete(
+                transactionDb.getDefaultNativeHandle(),
+                columnFamilyContext.getKeyBufferArray(),
+                columnFamilyContext.getKeyLength());
+          });
+    }
   }
 
   @Override
   public void deleteIfExists(final KeyType key) {
-    ensureInOpenTransaction(
-        transaction -> {
-          columnFamilyContext.writeKey(key);
-          transaction.delete(
-              transactionDb.getDefaultNativeHandle(),
-              columnFamilyContext.getKeyBufferArray(),
-              columnFamilyContext.getKeyLength());
-        });
+    try (final var timer = metrics.measureDeleteLatency()) {
+      ensureInOpenTransaction(
+          transaction -> {
+            columnFamilyContext.writeKey(key);
+            transaction.delete(
+                transactionDb.getDefaultNativeHandle(),
+                columnFamilyContext.getKeyBufferArray(),
+                columnFamilyContext.getKeyLength());
+          });
+    }
   }
 
   @Override
   public boolean exists(final KeyType key) {
-    ensureInOpenTransaction(
-        transaction -> {
-          columnFamilyContext.writeKey(key);
-          final byte[] value =
-              transaction.get(
-                  transactionDb.getDefaultNativeHandle(),
-                  transactionDb.getReadOptionsNativeHandle(),
-                  columnFamilyContext.getKeyBufferArray(),
-                  columnFamilyContext.getKeyLength());
-          columnFamilyContext.wrapValueView(value);
-        });
-    return !columnFamilyContext.isValueViewEmpty();
+    try (final var timer = metrics.measureGetLatency()) {
+      ensureInOpenTransaction(
+          transaction -> {
+            columnFamilyContext.writeKey(key);
+            final byte[] value =
+                transaction.get(
```

Comment:
We could also move all of these metrics measurement in the RocksInternal I guess? :thinking_face:

Reply:
`RocksDbInternal` doesn't do iteration and I feel like it fits better in `TransactionColumnFamily` because `RocksDbInternal` is more about exposing the native methods, not actually managing the reads and writes.